src/inputType/PinchInput.ts

```diff
@@ -203,17 +104,169 @@ export class PinchInput implements IInputType {
+			if ("PointerEvent" in window) {
+				this.element.addEventListener("pointerdown", this.onPinchStart, false);
+				this.element.addEventListener("pointermove", this.onPinchMove, false);
+				this.element.addEventListener("pointerup", this.onPinchEnd, false);
+				this.element.addEventListener("pointercancel", this.onPinchEnd, false);
+			} else if ("MSPointerEvent" in window) {
+				this.element.addEventListener("MSPointerDown", this.onPinchStart, false);
+				this.element.addEventListener("MSPointerMove", this.onPinchMove, false);
+				this.element.addEventListener("MSPointerUp", this.onPinchEnd, false);
+				this.element.addEventListener("MSPointerCancel", this.onPinchEnd, false);
+			}
+		} else {
+			this.element.addEventListener("touchstart", this.onPinchStart, false);
+			this.element.addEventListener("touchmove", this.onPinchMove, false);
+			this.element.addEventListener("touchend", this.onPinchEnd, false);
+			this.element.addEventListener("touchcancel", this.onPinchEnd, false);
+		}
+	}
+
+	private dettachEvent() {
+		if (this.activeInput === "pointer") {
+			if ("PointerEvent" in window) {
+				this.element.removeEventListener("pointerdown", this.onPinchStart, false);
+				this.element.removeEventListener("pointermove", this.onPinchMove, false);
+				this.element.removeEventListener("pointerup", this.onPinchEnd, false);
+				this.element.removeEventListener("pointercancel", this.onPinchEnd, false);
+			} else if ("MSPointerEvent" in window) {
+				this.element.removeEventListener("MSPointerDown", this.onPinchStart, false);
+				this.element.removeEventListener("MSPointerMove", this.onPinchMove, false);
+				this.element.removeEventListener("MSPointerUp", this.onPinchEnd, false);
+				this.element.removeEventListener("MSPointerCancel", this.onPinchEnd, false);
+			}
+		} else {
+			this.element.removeEventListener("touchstart", this.onPinchStart, false);
+			this.element.removeEventListener("touchmove", this.onPinchMove, false);
+			this.element.removeEventListener("touchend", this.onPinchEnd, false);
+			this.element.removeEventListener("touchcancel", this.onPinchEnd, false);
+		}
+		this.isEnabled = false;
+		this.observer = null;
+	}
+
+	private getTouches(event: InputEventType) {
+		if (event instanceof PointerEvent) {
+			return this.eventCache.length;
+		} else if (event instanceof TouchEvent) {
+			return event.touches.length;
+		}
+	}
+
+	private getOffset(pinchScale: number, prev: number = 1): number {
+		return this.baseValue * (pinchScale - prev) * this.options.scale;
+	}
+
+	private getScaleFromPointers() {
+		return this.getDistanceFromTouch(this.eventCache[0], this.eventCache[1]) / this.getDistanceFromTouch(this.firstPointers[0], this.firstPointers[1]);
+	}
+
+	private getScaleFromTouch(start: TouchList, end: TouchList) {
+		return this.getDistanceFromTouch(end[0], end[1]) / this.getDistanceFromTouch(start[0], start[1]);
+	}
+
+	private getDistanceFromTouch(p1: Touch | PointerEvent, p2: Touch | PointerEvent) {
```

Comment:
It would be better if a more meaningful naming such as startPoint than p1 and endPoint than p2.

Reply:
구현 당시 Pinch 제스쳐에서 두 손가락이 터치하는 두 개의 지점을 start와 end로 칭하면 혼란스러울 수 있을 것 같다고 판단하였습니다. 그렇지만 각각의 지점을 p1과 p2로 명명하는 것보다 좋은 이름이 있을지 생각해보겠습니다.

It has been decided that it might be confusing if we call the two points where two fingers touch in the pinch gesture as start and end.
But It should be considered if there's a better name than to call each point p1 and p2.